File: micropython/helper-classes/motor.py
from machine import PWM
from utime import sleep

class Motor():
    def __init__(self, A,B):
        self.A = PWM(A) # Creating a PWM object for motor phase A
        self.A.freq(500) # Establishing PWM frequency
        self.A.duty_u16(0) # Setting 0% duty cycle
        self.B = PWM(B) # Creating a PWM object for motor phase B
        self.B.freq(500) # Esablishing PWM frequency
        self.B.duty_u16(0) # Setting 0% duty cycle
        
    def u16(self,percent): # Helper function converting the speed to u16
        return int(percent * 65536) 
        
    def mbreak(self):
        self.A.duty_u16(0) # Set both PWM low in order to break
        self.B.duty_u16(0)
        
    def mfloat(self):
        self.A.duty_u16(self.u16(1)) #
        self.B.duty_u16(self.u16(1)) #

    def speed(self, vel):
        speed = self.u16(min(abs(vel/100),1)) # Define the speed of the motor
        a,b = (self.A,self.B) if vel < 0 else (self.B,self.A) # Reverse the direction if the comm
        a.duty_u16(0)  # to float between pulses, make this a 1
        b.duty_u16(speed)
        
    def pulse(self,vel,time):
        self.speed(vel)
        sleep(time)
        self.speed(0)

File: c++/motor.cpp
#include "Motor.h"
#include "PWM.h"
#include "pico/stdlib.h"

// Default constructor
Motor::Motor()
    : PIN_A(20), PIN_B(21), FREQ(500)
{
}

// Non-default constructor
Motor::Motor(uint PIN_A, uint PIN_B)
{
    this->PIN_A = PIN_A;
    this->A = PWM(PIN_A, FREQ);
    this->PIN_B = PIN_B;
    this->B = PWM(PIN_B, FREQ);
}

// Speed 
void Motor::speed(int vel)
{
    if (vel < 0) {
        A.duty(-1*vel);
        B.duty(0);
    } else {
        A.duty(0);
        B.duty(vel);
    }
}

void Motor::pulse(int vel, int pulse_ms)
{
    speed(vel); // Turn on for set duty cycle
    sleep_ms(pulse_ms); // sleep for the pulse time
    speed(0);   // Shut off after sleep
}